<?xml version="1.0"?>
<launch>
    
    <!-- Publish a pcl2 around the robot position at time t. -->
    <node name="signal_mapper" pkg="era_5g_network_signal_mapper" type="signal_mapper.py" output="screen">

        <!-- Base link frame of the robot -->
        <param name="base_link"   type="string" value="robot_base_link" />
	<!-- New frame where pcl2 will be publish -->
	<param name="semantic_map_frame"   type="string" value="semantic_map" />
	
    </node>

    <!-- Publish a concatenated (historical) pcl2 from the signal_mapper publisher pcl2. -->
    <node name="sub_signal_mapper" pkg="era_5g_network_signal_mapper" type="sub_signal_mapper.py" output="screen">

	<!-- Map frame -->
	<param name="map_frame"   type="string" value="robot_map" />
        <!-- New frame where pcl2 will be publish -->
        <param name="semantic_map_frame"   type="string" value="semantic_map" />
    </node>

    <!-- Publish a nav_msgs/OccupancyGrid msg from the pcl2 depending on the colour filtering-->
    <node name="costmap_translate" pkg="era_5g_network_signal_mapper" type="costmap_translate.py" output="screen">
        <!-- Map frame -->
	<param name="map_frame"   type="string" value="robot_map" />
        <!-- Topic of main map -->
        <param name="map_topic"   type="string" value="/robot/map" />
        <!-- Topic of main map metadata -->
        <param name="map_metadata_topic"   type="string" value="/robot/map_metadata" />
    </node>

</launch>
